### PR TITLE
Bump scriptworker version to 42.1.0

### PR DESCRIPTION
**Pull request by hneiva**

### CODE REVIEW

File: treescript/requirements/local.py38.txt [hunk 1]
```diff
@@ -7,4 +7,7 @@
 #
 -r test.py38.txt
 
-# The following packages are considered to be unsafe in a requirements file:
+# WARNING: The following packages were not pinned, but pip requires them to be
+# pinned when the requirements file includes hashes. Consider using the --allow-unsafe flag.
+# pip
+# setuptools
```
Comment (hneiva) [inline treescript/requirements/local.py38.txt:13]: @escapewindow is this a concern?

File: addonscript/requirements/base.py38.txt [hunk 2]
```diff
@@ -5,44 +5,79 @@
...
-    --hash=sha256:c6e9dcb4cb338d91a73f178d866d051efe7c62a7166653a91e7d9fb18274058f \
-    --hash=sha256:cffe3ab27871bc3ea47df5d8f7013945712c46a3cc5a95b6bee15887f1675c22 \
-    --hash=sha256:d012ad7911653a906425d8473a1465caa9f8dea7fcf07b6d870397b774ea7c0f \
-    --hash=sha256:d9e13b33afd39ddeb377eff2c1c4f00544e191e1d1dee5b6c51ddee8ea6f0cf5 \
-    --hash=sha256:e4b2b334e68b18ac9817d828ba44d8fcb391f6acb398bcc5062b14b2cbeac970 \
-    --hash=sha256:e54962802d4b8b18b6207d4a927032826af39395a3bd9196a5af43fc4e60b009 \
-    --hash=sha256:f705e12750171c0ab4ef2a3c76b9a4024a62c4103e3a55dd6f99265b9bc6fcfc \
-    --hash=sha256:f881853d2643a29e643609da57b96d5f9c9b93f62429dcc1cbb413c7d07f0e1a \
-    --hash=sha256:fe60131d21b31fd1a14bd43e6bb88256f69dfc3188b3a89d736d6c71ed43ec95
+aiohttp==3.8.1 \
```
Comment (escapewindow) [inline addonscript/requirements/base.py38.txt:8]: I think we're already using aiohttp 3.8.1 for pushmsix, so I'm less worried about this. aiohttp bumps can sometimes be prone to issues. https://docs.aiohttp.org/en/stable/changes.html

File: addonscript/requirements/base.py38.txt [hunk 3]
```diff
@@ -124,42 +163,34 @@ cffi==1.15.0 \
...
     # via
-    #   jwcrypto
-    #   scriptworker
-deprecated==1.2.13 \
-    --hash=sha256:43ac5335da90c31c24ba028af536a91d41d53f9e6901ddb021bcc572ce44e38d \
-    --hash=sha256:64756e3e14c8c5eea9795d93c524551432a0be75629f8f29e67ab8caf076c76d
-    # via jwcrypto
+    #   aiohttp
+    #   requests
+cryptography==36.0.1 \
```
Comment (escapewindow) [inline addonscript/requirements/base.py38.txt:172]: I *think* we'll be ok, with https://cryptography.io/en/latest/changelog/#v36-0-1 . Looks like 37.0.0 may need some checks before we upgrade.

File: balrogscript/requirements/base.py38.txt [hunk 4]
```diff
@@ -132,61 +171,117 @@ cffi==1.15.0 \
...
+    --hash=sha256:ebc15b1c22e55c4d5566e3ca4db8689470a0ca2babef8e3a9ee057a8b82ce4b1 \
+    --hash=sha256:ec63da4e7e4a5f924b90af42eddf20b698a70e58d86a72d943857c4c6045b3ee
     # via
-    #   jwcrypto
     #   mar
     #   scriptworker
-deepmerge==0.3.0 \
-    --hash=sha256:87166dbe9ba1a3348a45c9d4ada6778f518d41afc0b85aa017ea3041facc3f9c \
-    --hash=sha256:f6fd7f1293c535fb599e197e750dbe8674503c5d2a89759b3c72a3c46746d4fd
+deepmerge==1.0.1 \
```
Comment (escapewindow) [inline balrogscript/requirements/base.py38.txt:211]: I don't see anything concerning [here](https://github.com/toumorokoshi/deepmerge/compare/v0.3.0...v1.0.1), but I do wish they'd keep a changelog.

File: balrogscript/requirements/base.py38.txt [hunk 5]
```diff
@@ -132,61 +171,117 @@ cffi==1.15.0 \
...
+    --hash=sha256:e84cb61b0ac40a0c3e0e8b79c575161c5300d1d89e13c0e02f76193982f066ed \
+    --hash=sha256:e982878792c971cbd60ee510c4ee5bf089a8246226dea1f2138aa0bb67aff148 \
+    --hash=sha256:f20baa05eaa2bcd5404c445ec51aed1c268d62600362dc6cfe04fae34a424bd9 \
+    --hash=sha256:f7353ba3367473d1d616ee727945f439e027f0bb16ac1a750219a8344d1d5d3c \
+    --hash=sha256:f96293d6f982c58ebebb428c50163d010c2f05de0cde99fd681bfdc18d4b2dc2 \
+    --hash=sha256:ff9310f05b9d9c5c4dd472983dc956901ee6cb2c3ec1ab116ecdde25f3ce4951
+    # via
+    #   aiohttp
+    #   aiosignal
+github3-py==3.1.0 \
```
Comment (escapewindow) [inline balrogscript/requirements/base.py38.txt:282]: I think branch protections are the only incompatible change in this [major version bump](https://github.com/sigmavirus24/github3.py/blob/main/docs/source/release-notes/3.0.0.rst) ?

**Review by escapewindow — APPROVED** [review]
Spot checked.

This is a decent sized change, so:

- [scriptworker_canary](https://searchfox.org/mozilla-central/source/taskcluster/gecko_taskgraph/actions/scriptworker_canary.py) can help us find issues after rollout;
- we can try a [retrigger force](https://docs.mozilla-releng.net/en/latest/taskcluster/rerun_vs_retrigger.html#testing-scriptworker-pools-outside-of-the-release-retrigger-force) to test some signing tasks after the fact;
- if we're worried, we can deploy to dev, change all of the worker types to -dev, and run a try staging release;
- ideally we find a quiet time where we can monitor what's going on and be ready to roll back or fix forward if needed.

File: treescript/requirements/local.py38.txt (hunk 1 above)
Comment (escapewindow) [reply treescript/requirements/local.py38.txt:13]: Might be. If it's a problem, we may need to change pin.sh to --allow-unsafe; not sure if that'll be for all or just for local or what.

**Comment by escapewindow** [comment]
Looks like scriptworker_client is failing black linting?
Not sure if this was broken before, or due to a black version bump.

```
[task 2022-02-17T00:09:31.149Z] =================================== FAILURES ===================================
[task 2022-02-17T00:09:31.150Z] ______________________________ Black format check ______________________________
[task 2022-02-17T00:09:31.150Z] --- /builds/worker/checkouts/vcs/scriptworker_client/tests/test_utils.py	2022-02-17 00:00:55.987617 +0000
[task 2022-02-17T00:09:31.151Z] +++ /builds/worker/checkouts/vcs/scriptworker_client/tests/test_utils.py	2022-02-17 00:09:30.725920 +0000
[task 2022-02-17T00:09:31.152Z] @@ -50,22 +50,19 @@
[task 2022-02-17T00:09:31.152Z]  
[task 2022-02-17T00:09:31.152Z]  # helpers {{{1
[task 2022-02-17T00:09:31.152Z]  # from https://github.com/SecurityInnovation/PGPy/blob/develop/tests/test_01_types.py
[task 2022-02-17T00:09:31.153Z]  text = {  # some basic utf-8 test strings - these should all pass             'english': u'The quick brown fox jumped over the lazy dog',
[task 2022-02-17T00:09:31.154Z]      # this hiragana pangram comes from http://www.columbia.edu/~fdc/utf8/
[task 2022-02-17T00:09:31.154Z] -    "hiragana": u"いろはにほへど　ちりぬるを\n"
[task 2022-02-17T00:09:31.155Z] -    u"わがよたれぞ　つねならむ\n"
[task 2022-02-17T00:09:31.155Z] -    u"うゐのおくやま　けふこえて\n"
[task 2022-02-17T00:09:31.156Z] -    u"あさきゆめみじ　ゑひもせず",
[task 2022-02-17T00:09:31.156Z] -    "poo": u"Hello, \U0001F4A9!",
[task 2022-02-17T00:09:31.157Z] +    "hiragana": "いろはにほへど　ちりぬるを\n" "わがよたれぞ　つねならむ\n" "うゐのおくやま　けふこえて\n" "あさきゆめみじ　ゑひもせず",
[task 2022-02-17T00:09:31.157Z] +    "poo": "Hello, \U0001F4A9!",
[task 2022-02-17T00:09:31.158Z]  }
[task 2022-02-17T00:09:31.158Z]  
[task 2022-02-17T00:09:31.158Z]  non_text = {
[task 2022-02-17T00:09:31.158Z]      "None": None,
[task 2022-02-17T00:09:31.159Z]      "dict": {"a": 1, 2: 3},
[task 2022-02-17T00:09:31.159Z] -    "cyrillic": u"грызть гранит науки".encode("iso8859_5"),
[task 2022-02-17T00:09:31.159Z] -    "cp865": u"Mit luftpudefartøj er fyldt med ål".encode("cp865"),
[task 2022-02-17T00:09:31.160Z] +    "cyrillic": "грызть гранит науки".encode("iso8859_5"),
[task 2022-02-17T00:09:31.160Z] +    "cp865": "Mit luftpudefartøj er fyldt med ål".encode("cp865"),
```

**Pull request merged**